htlcswitch/interceptable_switch.go

```diff
@@ -93,6 +94,11 @@ type InterceptableSwitch struct {
 	// currentHeight is the currently best known height.
 	currentHeight int32
 
+	// signChannelUpdate is used when an intercepting application includes
+	// an unsigned channel update to be signed by us.
+	signChannelUpdate func(u *lnwire.ChannelUpdate) (*ecdsa.Signature,
```

Comment:
Hmm, shouldn't we assume that they want to instead send the latest update? Otherwise i can see this causing an error where: user sends custom channel update, the one on disk conflicts, so routing errors occur.

Reply:
We can prevent this in the forwarding level during the validation of the returned message. If it's an encoded channel update, then it needs to match what we'd send out anyway.

Reply:
The context in which this is to be used is to update the routing policy of a virtual channel. The channel isn't broadcasted through gossip and doesn't exist in the graph on disk. There is no 'latest update'. 

The existence of the virtual channel is solely communicated to the payer through invoice route hints. If the receiver (the virtual hop) changes its policy, the channel update contained in the failure message updates the sender with the new values.

Reply:
> The context in which this is to be used is to update the routing policy of a virtual channel. The channel isn't broadcasted through gossip and doesn't exist in the graph on disk.

Assuming the virtual channel has a distinct pubkey (and also shared secret in the onion), then how would this work in practice? lnd would use diff logic for if it was the final hop vs the penultimate hop.

Reply:
If node N is the virtual node, then it is the job of node N-1 to return a channel update. The example is again `fee_insufficient`.

Real node N-1 doesn't know about this channel at all (it is virtual), so what happens is that the virtual node N returns a channel update for the virtual channel to N-1, and N-1 just signs the update.